src/Physics/NuElectron/XSection/PXSecOnElectron.cxx

```diff
@@ -0,0 +1,118 @@
+//____________________________________________________________________________
+/*
+ Copyright (c) 2003-2023, The GENIE Collaboration
+ For the full text of the license visit http://copyright.genie-mc.org
+
+ Brinden Carlson bcarlson1@ufl.edu
+ University of Florida
+*/
+//____________________________________________________________________________
+
+#include "Framework/Algorithm/AlgConfigPool.h"
+#include "Framework/Conventions/GBuild.h"
+#include "Framework/Conventions/Controls.h"
+#include "Physics/XSectionIntegration/XSecIntegratorI.h"
+#include "Framework/Conventions/Constants.h"
+#include "Framework/Conventions/RefFrame.h"
+#include "Physics/NuElectron/XSection/PXSecOnElectron.h"
+#include "Framework/Messenger/Messenger.h"
+#include "Framework/ParticleData/PDGUtils.h"
+#include "Framework/Utils/KineUtils.h"
+#include "Physics/NuclearState/ElectronVelocity.h"
+
+#include "TFile.h"
+#include "TGraph.h"
+#include <fstream>
+#include <iterator>
+
+using namespace genie;
+using namespace genie::constants;
+using namespace genie::controls;
+
+//____________________________________________________________________________
+PXSecOnElectron::PXSecOnElectron(string name, string config) :
+XSecAlgorithmI(name, config)
+{
+
+}
+//____________________________________________________________________________
+PXSecOnElectron::PXSecOnElectron() :
+XSecAlgorithmI("genie::PXSecOnElectron")
+{
+
+}
+//____________________________________________________________________________
+PXSecOnElectron::~PXSecOnElectron()
+{
+
+}
+//____________________________________________________________________________
+double PXSecOnElectron::Integral(const Interaction * interaction) const
+{
+  double xsec_sum = 0; 
+  double xsec_sum2 = 0; 
+  int NInt = 0; //Count number of integrations
+  do{
+    NInt++;
+    Interaction in_curr(*interaction); //Copy interaction object
+    fElectronVelocity->InitializeVelocity(in_curr); //Modify interaction to give electron random velocity from selected distribution
+    double xsec = fXSecIntegrator->Integrate(this,&in_curr); // In ele at rest
+    //get beta - comps orthogonal to beam x,y
+    //scale = sqrt(1-b_t^2)
+    TVector3 beta = in_curr.InitState().Tgt().HitPartP4().BoostVector(); // beta
+    double beta_tangent = sqrt(TMath::Power(beta[0],2)+TMath::Power(beta[1],2)); //Component tangential to beam
```

Comment:
This is wrong. What we need here is a the component of beta orthogonal to the probe. Since the probe is not always parallel to z this math needs to be rewritten